test/taxCalculator.test.js

```diff
@@ -1,57 +1,87 @@
 const calculateTaxes = require('../src/taxCalculator.js')
+const Big = require('big.js');
 
 const assert = require('assert');
 const test = require('mocha').it;
 
 describe('Calculate taxes', function () {
 
-    test('It applies tax rate 1 only', function () {
+    test('It should return taxes correctly when taxableIncome is less than or equal to zero', async function () {
+        // taxableIncome <= 5m (tax level 1)
+        const taxableIncome = -1;
+
+        const expectedTaxes = {
+            rates: [],
+            totalTax: new Big(0)
+        };
+
+        await calculateTaxes(taxableIncome)
+            .then(function (actual) {
+                assert.deepStrictEqual(actual, expectedTaxes)
+
+            }).catch(function () {
+            })
+    })
+
+    test('It applies tax rate 1 only', async function () {
         // taxableIncome <= 5m (tax level 1)
 
-        const actualTaxes = calculateTaxes(1_234_444);
         const expectedTaxes = {
             rates: [
                 { name: 'Tax level 1', rate: 0.05, amount: 61722.2 },
             ],
             totalTax: 61722.2
         };
-        assert.deepStrictEqual(actualTaxes, expectedTaxes);
+
+        await calculateTaxes(1_234_444)
+            .then(function (actual) {
+                assert.deepStrictEqual(actual, expectedTaxes)
+
+            }).catch(function () {
+
+            })
     })
-    test('It applies 2 first tax rates', function () {
+
+    test('It applies 2 first tax rates', async function () {
         // taxableIncome <= 5m (tax level 1) +  5m (tax level 2) 
         //5m < taxableIncome <= 10m
 
-        const actualTaxes = calculateTaxes(9_000_000);
         const expectedTaxes = {
             rates: [
                 { name: 'Tax level 1', rate: 0.05, amount: 250_000 },
                 { name: 'Tax level 2', rate: 0.1, amount: 400_000 },
             ],
             totalTax: 650_000
         };
-        assert.deepStrictEqual(actualTaxes, expectedTaxes);
+
+        await calculateTaxes(9_000_000).then(function (taxes) {
+            assert.deepStrictEqual(taxes, expectedTaxes)
+        })
     })
 
-    test('It applies 3 first tax rates', function () {
+    test('It applies 3 first tax rates', async function () {
         // 10m < taxableIncome <= 5m (tax level 1) +  5m (tax level 2) + 8m (tax level 3)
         //10m < taxableIncome <= 18m
-
-        const actualTaxes = calculateTaxes(12_000_000);
         const expectedTaxes = {
             rates: [
                 { name: 'Tax level 1', rate: 0.05, amount: 250_000 },
                 { name: 'Tax level 2', rate: 0.1, amount: 500000 },
                 { name: 'Tax level 3', rate: 0.15, amount: 300000 }],
             totalTax: 1050000
         };
-        assert.deepStrictEqual(actualTaxes, expectedTaxes);
+
+        await calculateTaxes(12_000_000).then(function (taxes) {
+
```

Comment:
why do we have so many redundant spaces here?

Reply:
just 1 line